### PR TITLE
Unittest for Repartition

### PR DESCRIPTION
**Pull request by valiantljk**

### CODE REVIEW

File: deltacat/compute/compactor/steps/repartition.py [hunk 1]
```diff
@@ -43,13 +43,79 @@ class RepartitionType(str, Enum):
...
 
 
+def repartition_range(
+    tables: List[pa.Table],
+    destination_partition: Partition,
+    repartition_args: dict,
+    max_records_per_output_file: int,
+    repartitioned_file_content_type: ContentType = ContentType.PARQUET,
+    deltacat_storage=unimplemented_deltacat_storage,
+):
```
Comment (pfaraone) [inline deltacat/compute/compactor/steps/repartition.py:60]: Can you add a docstring here?

File: deltacat/compute/compactor/steps/repartition.py [hunk 2]
```diff
@@ -43,13 +43,79 @@ class RepartitionType(str, Enum):
...
+                    (pc.field(col_name_int64) > pc.scalar(lower_limit))
+                    & (pc.field(col_name_int64) <= pc.scalar(upper_limit))
+                )
+            )
+    partition_table_length = 0
+    # After re-grouping the tables by specified ranges, for each group, we need concat and stage the tables
+    partition_deltas: List[Delta] = []
+    for partition_tables in partitioned_tables_list:
+        if len(partition_tables) > 0:
+            partition_table = pa.concat_tables(partition_tables)
```
Comment (pfaraone) [inline deltacat/compute/compactor/steps/repartition.py:100]: Can you add type hints in this block?

**Comment by Zyiqin-Miranda** [comment]
LGTM!

File: deltacat/compute/compactor/steps/repartition.py [hunk 3]
```diff
@@ -43,13 +43,103 @@ class RepartitionType(str, Enum):
...
+    """
+    column: str = repartition_args["column"]
+    partition_ranges: List = repartition_args["ranges"]
+    if len(partition_ranges) == 0:
+        raise ValueError("No partition ranges specified")
+    # check if the column exists in the table
+    # TODO: design a better way to handle the case when the column does not exist in the table, e.g., backfill + repartition by stream position + file id
+    if not all(column in table.column_names for table in tables):
+        raise ValueError(f"Column {column} does not exist in the table")
+    # given a range [x, y, z], we need to split the table into 4 files, i.e., (-inf, x], (x, y], (y, z], (z, inf)
```
Comment (Zyiqin-Miranda) [inline deltacat/compute/compactor/steps/repartition.py:93]: Could we move this comment up in doc string as a full example to show clearly it's open bounded on lower limit and closed bounded on upper limit. 
Like example args has 3 values [x,y,z] which will split the table into (-inf, x], (x, y], (y, z], (z, inf) 4 ranges?

**Review by rkenmi — APPROVED** [review]

**Review by Zyiqin-Miranda — APPROVED** [review]

**Review by raghumdani — APPROVED** [review]
LGTM

**Pull request merged**